### PR TITLE
Create basic HTTP Purger UI

### PR DESCRIPTION
**Pull request by anavarre**

### CODE REVIEW

File: modules/purge_purger_http/src/Form/ConfigurationForm.php [hunk 1]
```diff
@@ -33,25 +40,79 @@ public function getFormID() {
    * {@inheritdoc}
    */
   public function buildForm(array $form, FormStateInterface $form_state) {
-    $form['test'] = [
+
```
Comment (nielsvm) [inline modules/purge_purger_http/src/Form/ConfigurationForm.php]: Although even I find it look nice that way, code in the method body (within {}) should be sticked straight to the "public function" line. In other words, delete line 43

File: modules/purge_purger_http/src/Form/ConfigurationForm.php [hunk 2]
```diff
@@ -33,25 +40,79 @@ public function getFormID() {
...
+      '#default_value' => $config->get('hostname'),
+      '#required' => FALSE,
+    ];
+    $form['http_settings']['port'] = [
+      '#title' => $this->t('Port'),
+      '#type' => 'textfield',
+      '#default_value' => $config->get('port'),
+      '#required' => FALSE,
+    ];
+    /*
```
Comment (nielsvm) [inline modules/purge_purger_http/src/Form/ConfigurationForm.php]: newline above that docblock please, perhaps even below it.

File: modules/purge_purger_http/src/Form/ConfigurationForm.php [hunk 3]
```diff
@@ -33,25 +40,79 @@ public function getFormID() {
...
+     * @todo We should get token support in the future.
+     */
+    $form['http_settings']['path'] = [
+      '#title' => $this->t('Path'),
+      '#type' => 'textfield',
+      '#default_value' => $config->get('path'),
+      '#required' => FALSE,
+    ];
+    /*
```
Comment (nielsvm) [inline modules/purge_purger_http/src/Form/ConfigurationForm.php:74]: newline above that docblock please, perhaps even below it.

File: modules/purge_purger_http/src/Form/ConfigurationForm.php [hunk 4]
```diff
@@ -33,25 +40,79 @@ public function getFormID() {
...
     ];
-    return parent::buildForm($form, $form_state);
-  }
 
-  /**
-   * {@inheritdoc}
-   */
-  public function validateForm(array &$form, FormStateInterface $form_state) {
-    $form_state->setError($form['test'], $this->t('Not yet implemented yet.'));
+    /*
```
Comment (nielsvm) [inline modules/purge_purger_http/src/Form/ConfigurationForm.php]: newline above that docblock please, perhaps even below it.

File: modules/purge_purger_http/src/Form/ConfigurationForm.php [hunk 5]
```diff
@@ -33,25 +40,79 @@ public function getFormID() {
...
+    $this->config('purge_purger_http.settings')
+      ->set('hostname', $form_state->getValue('hostname'))
+      ->set('port', $form_state->getValue('port'))
+      ->set('path', $form_state->getValue('path'))
+      ->set('request_method', $this->request_methods[$form_state->getValue('request_method')])
+      ->save();
+
     return parent::submitForm($form, $form_state);
   }
 }
```
Comment (nielsvm) [inline modules/purge_purger_http/src/Form/ConfigurationForm.php:116]: newline above last closing curly brace (drupal policy), so one line between 117-118

File: modules/purge_purger_http/src/Form/ConfigurationForm.php (hunk 1 above)
Comment (anavarre) [inline modules/purge_purger_http/src/Form/ConfigurationForm.php]: Fixed.

File: modules/purge_purger_http/src/Form/ConfigurationForm.php (hunk 2 above)
Comment (anavarre) [inline modules/purge_purger_http/src/Form/ConfigurationForm.php]: Missed it. Fixed.

File: modules/purge_purger_http/src/Form/ConfigurationForm.php (hunk 3 above)
Comment (anavarre) [inline modules/purge_purger_http/src/Form/ConfigurationForm.php:74]: Fixed.

File: modules/purge_purger_http/src/Form/ConfigurationForm.php (hunk 5 above)
Comment (anavarre) [inline modules/purge_purger_http/src/Form/ConfigurationForm.php:116]: Dang, it was there at some point. Dunno how it ended up not being there on commit. Thanks, fixed.

File: modules/purge_purger_http/src/Form/ConfigurationForm.php (hunk 4 above)
Comment (anavarre) [inline modules/purge_purger_http/src/Form/ConfigurationForm.php]: Fixed.

**Pull request merged**